evio/src/main/java/org/hps/evio/TriggerConfigEvioReader.java

```diff
@@ -39,28 +44,54 @@ public void getDAQConfig(EvioEvent evioEvent, EventHeader lcsimEvent) {
             }
             return;
         }
-        List<EvioDAQParser> trigconf = new ArrayList<EvioDAQParser>();
-        for (BaseStructure bank : evioEvent.getChildrenList()) {
-            if (bank.getChildCount() <= 0)
-                continue;
-            int crate = bank.getHeader().getTag();
-            for (BaseStructure subBank : bank.getChildrenList()) {
-                if (subBank.getHeader().getTag() == EvioDAQParser.BANK_TAG) {
-                    if (subBank.getStringData() == null) {
-                        Logger.getLogger(this.getClass().getName()).log(Level.SEVERE,
-                                "JEVIO can't parse DAQ Config bank.  Event Number "
-                                        + evioEvent.getEventNumber());
-                        return;
+        else if (lcsimEvent.getRunNumber() >= 4043 && lcsimEvent.getRunNumber() < 9000) {
```

Comment:
I'd suggest you don't hard-code checks on run numbers like this.  Instead, update this class with a flag or other configuration based on the conditions system hooks.

Reply:
Do you mean that I set a flag according to run number ranges, and then put expression with flag into "if condition"?
What kind of condition parameters/hooks can be used to identify run number ranges?

Reply:
Ah... I think I got what you said. You mean use like DatabaseConditionsManager.isPhys2019Run(lcsimEvent.getRunNumber())
instead of 9000

Reply:
Whatever class is creating this one (event builder?) should have a `detectorChanged` or other conditions system hook where you can set some configuration on the parser.  I would either set a boolean flag or better yet, just have a small class that implements the correct behavior, e.g. register a specific object depending on what run number is seen in the conditions callback method.  We try not to hard-code specific run numbers within our algorithms.

If what you have works, then we can merge without doing this.  I guess for now it is okay if we just want to get something working.  (I see this has already been approved so I assume it works.)

Reply:
Not exactly...

In `LCSimEngRunEventBuilder`, there is a method `conditionsChanged` where you should configure your class depending on what run number is seen there.  You can get the run number from the manager there.

This keeps all the conditions configuration/checking in one place and not scattered around in many places within the code.

In fact, shouldn't there be an entirely separate `TriggerConfigEvioReader` class for 2019? 

I don't think it is a great idea to modify the one that's there with a bunch of if/then statements depending on the run number.  It would be better to have entirely separate top-level classes.

Reply:
Agree. Could I merge this branch now since we need this parser for some studies of trigger validation? And then I will build a new branch to make changes according to your suggestions.

Reply:
The changes will not affect application of the parser.